### PR TITLE
Added schema to users route

### PR DESCRIPTION
**Pull request by dalebremner**
@RackHD/corecommitters @srinia6 @BillyAbildgaard @keedya

### CODE REVIEW

File: static/schemas/2.0/user.2.0.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            "properties": {
+                "username": {
+                    "description": "The username of the user",
+                    "type": "string"
+                },
+                "password": {
+                    "description": "The password of the user",
+                    "type": "string"
+                },
+                "role": {
```
Comment (yyscamper) [inline static/schemas/2.0/user.2.0.json]: The `role` is to specify the read/write right? If so, change the role to `enum` will be better.

File: static/schemas/2.0/user.2.0.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            "description": "A RackHD local user",
+            "type": "object",
+            "properties": {
+                "username": {
+                    "description": "The username of the user",
+                    "type": "string"
+                },
+                "password": {
+                    "description": "The password of the user",
+                    "type": "string"
```
Comment (yyscamper) [inline static/schemas/2.0/user.2.0.json]: Recommend to add more conditions for `username` and `password`, these conditions will be played as validation code and keep the code & data clean.  Such as use `minLength` to specify the minimum characters for that filed, and use `patter` to screen some invalid characters.

File: static/schemas/2.0/user.2.0.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                    "description": "The role of the user",
+                    "type": "string"
+                }
+            },
+            "additionalProperties": false
+        },
+        "User": {
+            "allOf": [
+                { "$ref": "#/definitions/PartialUser" },
+                { "required": [ "username", "password" ] }
```
Comment (yyscamper) [inline static/schemas/2.0/user.2.0.json]: Not a big problem here, your code still works well here, but I suggest moving the `required` to within `PartialUser`, as they are strongly coupled with the PartialUser properties.

Comment (dalebremner) [inline static/schemas/2.0/user.2.0.json]: @yyscamper PartialUser is used for PATCH, so the user must be able to specify a variable number of properties using PartialUser.

File: static/schemas/2.0/user.2.0.json (hunk 1 above)
Comment (dalebremner) [inline static/schemas/2.0/user.2.0.json]: @yyscamper good idea. Will add the enum...

File: static/schemas/2.0/user.2.0.json (hunk 2 above)
Comment (dalebremner) [inline static/schemas/2.0/user.2.0.json]: @yyscamper agreed. I'll add more validation

File: static/schemas/2.0/user.2.0.json (hunk 3 above)
Comment (yyscamper) [inline static/schemas/2.0/user.2.0.json]: You are right, I didn't realized the PATH

File: static/schemas/2.0/user.2.0.json (hunk 1 above)
Comment (brianparry) [inline static/schemas/2.0/user.2.0.json]: @dalebremner @yyscamper @keedya Long term plan is to allow roles to be added, removed or modified, so the set of enumerated values could change at run-time.  "Read", "Write", etc are privileges not roles and it makes sense to enumerate them in the schema for the roles route when it is available.

Comment (yyscamper) [inline static/schemas/2.0/user.2.0.json]: If `role` can be updated or removed then I agree it's not suitable to make it as enum in schema.

**Comment by yyscamper** [comment]
:+1:

**Comment by brianparry** [comment]
Looks good :+1:

**Pull request merged**